CMakeLists.txt

```diff
@@ -1,21 +1,124 @@
 cmake_minimum_required(VERSION 3.16.0 FATAL_ERROR) # Configurable policies: <= CMP0097
 
-cmake_policy(SET CMP0091 NEW)
-cmake_policy(SET CMP0092 NEW)
-
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
-include(LibtorrentMacros)
-
-read_version("${CMAKE_CURRENT_SOURCE_DIR}/include/libtorrent/version.hpp" VER_MAJOR VER_MINOR VER_TINY)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
 
 project(libtorrent
-	DESCRIPTION "Bittorrent library"
-	VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_TINY}
+	VERSION "1.2.14"
+	DESCRIPTION "An efficient feature complete C++ bittorrent implementation"
+	HOMEPAGE_URL "https://libtorrent.org"
+	LANGUAGES C CXX
 )
-set (SOVERSION "10")
 
 include(GNUInstallDirs)
 include(GeneratePkgConfig)
+include(LibtorrentMacros)
+
+# This must be set before creating targets. Ignored on non-MSVC.
+set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>$<$<NOT:$<BOOL:${static_runtime}>>:DLL>")
+
+# -- Features --
+feature_option(NAME build_tests DESCRIPTION "build tests" DEFAULT OFF)
```

Comment:
Hereinafter: I believe names are used in CMake only for optional parameters.

Reply:
@zeule The point of using `cmake_parse_arguments()` like this instead of classic argument handling is to reap the following benefits:

- improved readability and self-documentation at the call site;
- the order in which the arguments are given doesn't matter;
- more robustness vs. manual argument parsing, and covers all of CMake's corner cases with function arguments.

Still I should probably better defend against the case of parameters that are actually required not being present by checking `<prefix>_KEYWORDS_MISSING_VALUES`.

Reply:
> improved readability and self-documentation at the call site

Yes, but a single comment line on top of the block will do the trick without repetitive keywords 

> the order in which the arguments are given doesn't matter;

So what?

> more robustness vs. manual argument parsing, and covers all of CMake's corner cases with function arguments.

There were no manual parsing. This is a function with constant number of arguments. Nothing requires sophisticated parsing there.